events/app/app.go

```diff
@@ -0,0 +1,46 @@
+// Package app provides an empire.EventStream implementation that publishs
+// events to the app's kinesis stream if available.
+package app
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/remind101/empire"
+)
+
+type kinesisClient interface {
+	PutRecord(*kinesis.PutRecordInput) (*kinesis.PutRecordOutput, error)
+}
+
+// EventStream is an implementation of the empire.EventStream interface backed
+// by Kinesis.
+type EventStream struct {
+	empire  *empire.Empire
+	kinesis kinesisClient
+}
+
+type Event interface {
```

Comment:
Minor, but might be nice to move this interface below https://github.com/remind101/empire/blob/master/events.go#L191-L198 as:

```go
// An AppEvent is an Event that relates to a specific app.
type AppEvent interface {
        Event
        App() *App // AppID is fine too.
}
```